src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultEnumerableConverter.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Text.Json.Serialization.Policies;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonEnumerableT<T> : ICollection<T>, IEnumerable<T>, IList<T>, IReadOnlyCollection<T>, IReadOnlyList<T>
+    {
+        List<T> _list;
+
+        public JsonEnumerableT(IList sourceList)
+        {
+            // TODO: Change sourceList from IList to List<T> so we can do a direct assignment here.
```

Comment:
Yes thanks for noting this. As we discussed I'm working on some perf improvements here to avoid boxing.